src/java/htsjdk/samtools/BAMFileReader.java

```diff
@@ -97,11 +100,13 @@
     BAMFileReader(final InputStream stream,
                   final File indexFile,
                   final boolean eagerDecode,
+                  final boolean useAsynchronousIO,
```

Comment:
maybe but the problem is this: the Asynch decompression implemented in https://github.com/samtools/htsjdk/pull/576 creates an asynch stream **in the constructor** of `BAMFileReader` so calling a setter is too late.